cli/azd/pkg/infra/provisioning/provider.go

```diff
@@ -19,9 +19,10 @@ const (
 )
 
 type Options struct {
-	Provider ProviderKind `yaml:"provider,omitempty"`
-	Path     string       `yaml:"path,omitempty"`
-	Module   string       `yaml:"module,omitempty"`
+	Provider ProviderKind   `yaml:"provider,omitempty"`
+	Path     string         `yaml:"path,omitempty"`
+	Module   string         `yaml:"module,omitempty"`
+	Config   map[string]any `yaml:"config,omitempty"`
```

Comment:
There are some tricks we can play in both JSON Schema for the YAML LSP, and go marshal/unmarshalling to avoid an extra `config` node in YAML for known union types. I can take a half hour to pull it out of my feature implementation [PR](https://github.com/Azure/azure-dev-pr/pull/1673/files#diff-b53ec5cd2fa533b7cc3ca2ff6715111a330bac7ba080efa8ec6cab3382ef4ea7), but just wanting to see a pulse if there's any interests here from a DX perspective.

So instead of:
```yaml
infra:
  provider: bicep
  config:
    deploymentStacks:
```

We would have:
```yaml
infra:
  provider: bicep
  deploymentStacks:
```

@ellismg @wbreza @vhvb1989

Reply:
But I see that we're imagining these are "dynamic passthrough information", so perhaps what we have here suffices.

Reply:
I do actually feel like the `config` node works well for these provider based configurations but since the `infra` section existed before adopted this model I am happy to not introduce the `config` node here.

I was able to remove the extra layer without anything to fancy.